src/1_utils.js

```diff
@@ -934,3 +934,15 @@ utils.convertObjectValuesToString = function(objectToConvert) {
 	return objectToConvert;
 };
 
+// Adds user-supplied metadata object from init to additionalMetadata['hosted_deeplink_data'] for Journeys targeting
+utils.addMetadataFromInitToHostedMetadata = function(metadataFromInit, additionalMetadata) {
```

Comment:
I would call this `mergeHostedDeeplinkData`. merge makes sense since that's what it's actually doing.  hosted_deeplink_data is what is actually getting merged in. This function is also used in `track` so using init is not accurate

Reply:
I would also have `additionalMetadata` as the first argument to be more consistent with out merge function which has the `to` object first

Reply:
Could you also return the results? I would prefer to make this a pure function with no side effects